### PR TITLE
feat: Add Global and Service Specific Context Path Capibility to all Novu Apps

### PR DESCRIPTION
**Pull request by Cliftonz**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
I have added support for context path separation of each service on the same URL.
This will allow each service to be set with a global context path and a service-specific context path.
For example, if the global context path is ```novu``` and the web context path is ```web``` then the path for authentication would be ```novu.co/novu/web/auth/user```.

This new feature defaults to no context path unless one of the following is set:
- GLOBAL_CONTEXT_PATH
- API_CONTEXT_PATH
- WIDGET_CONTEXT_PATH
- WS_CONTEXT_PATH
- WEB_CONTEXT_PATH

These variable names can easily be changed as they are only referenced in libs/shared/src/config/contextPath.ts 

- **Why was this change needed?** (You can also link to an open issue here)
This change was needed to create a helm/k8 deployment with reverse-proxy ingress controllers like Nginx, traefik, or app-gateway.

### CODE REVIEW

**Review by p-fernandez — COMMENTED** [review]
Thanks for your first contribution! 🍾 
I left some initial suggestions and comments to it.
Leave to you the overall approval, @scopsy.

File: libs/shared/src/config/contextPath.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        contextPath += process.env.FRONT_BASE_CONTEXTPATH + '/';
+      }
+      break;
+    case NovuComponentEnum.WIDGET:
+      if (process.env.WIDGET_CONTEXTPATH) {
+        contextPath += process.env.WIDGET_CONTEXTPATH + '/';
+      }
+      break;
+    case NovuComponentEnum.EMBEDED:
+      if (process.env.WIDGET_EMBED_CONTEXTPATH) {
```
[the remark this answers is not in the document]
Comment (Cliftonz) [reply libs/shared/src/config/contextPath.ts:33]: This container is deployed as part of the docker compose files, as from my understanding the embeded container is just a proxy, I agree that this is not needed.

File: apps/ws/src/.env.development [hunk 2]
```diff
@@ -5,3 +5,6 @@ REDIS_PORT=6379
 REDIS_HOST=localhost
 REDIS_DB_INDEX=2
 JWT_SECRET=TEST_CHANGE9s
+
+GLOBAL_CONTEXTPATH=
+REACT_APP_WS_CONTEXTPATH=
```
[the remark this answers is not in the document]
Comment (Cliftonz) [reply apps/ws/src/.env.development:10]: Sounds good, I also changed the web context path from FRONT_BASE_CONTEXTPATH to WEB_CONTEXT_PATH.

**Review by scopsy — APPROVED** [review]
Looks great @Cliftonz and thanks @p-fernandez for the great suggestions. We can merge this after tests will pass 🚀

**Pull request merged**